### PR TITLE
Updated Ids processing in Application Gateways

### LINKED ISSUE #3800: Set-AzureRmApplicationGateway throws "Object reference not set to an instance of an object" the first time it runs after "Add-AzureRmApplicationGatewayHttpListener"
### Cmdlet(s)
$appgateway = Add-AzureRmApplicationGatewayHttpListener -ApplicationGateway $appgw -Name $backendAddressPoolName -FrontendIPConfiguration $fipconfig01 -FrontendPort $fp01 `
												      -Protocol Https -HostName $hostName -RequireServerNameIndication true -SslCertificate $cert
            
 $UpdatedAppGw = Set-AzureRmApplicationGateway -ApplicationGateway $appgateway
### PowerShell Version
5.1.14393.953    

### OS Version
10.0.14393.953    

### Description
I am trying to add new HTTP listener to already existing Azure Application Gateway. After successfully running the cmdlet "Add-AzureRmApplicationGatewayHttpListener", the application gateway needs to be updated by running the cmdlet "Set-AzureRmApplicationGateway". It is here where the cmdlet throws "Object reference not set to an instance of an object" error the first time it is run. On retry, it successfully gets updated. I have tried the process repeatedly by deleting and adding the listeners, and this scenario is seen on every try. 
### Debug Output

Set-AzureRmApplicationGateway : Object reference not set to an instance of an object.
At line:94 char:29
+ ... atedAppGw = Set-AzureRmApplicationGateway -ApplicationGateway $appgat ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Set-AzureRmApplicationGateway], NullReferenceException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Network.SetAzureApplicationGatewayCommand

### PR DESCRIPTION
**Pull request by EvgenyAgafonchikov**
## Description

Fixes #3800

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by MikhailTryakhov** [comment]
@EvgenyAgafonchikov why is it not in a monthly branch?

**Comment by EvgenyAgafonchikov** [comment]
@cormacpayne 
- Updated base branch
- Do we need to update changelog for this change as it is mostly internal resources processing which is not exposed to users?
- In progress

File: src/Network/Network.Test/ScenarioTests/ApplicationGatewayTests.ps1 [hunk 1]
```diff
@@ -1465,12 +1465,8 @@ function Test-ApplicationGatewayCRUDSubItems2
 		# Add to test Remove
 		Add-AzApplicationGatewayBackendHttpSettings -ApplicationGateway $appgw -Name $poolSetting02Name -Port 1234 -Protocol Http -CookieBasedAffinity Enabled -RequestTimeout 42 -HostName test -Path /test -AffinityCookieName test
 		$fipconfig = Get-AzApplicationGatewayFrontendIPConfig -ApplicationGateway $appgw -Name $fipconfigName
-		$fp02 = Get-AzApplicationGatewayFrontendPort -ApplicationGateway $appgw -Name $frontendPort02Name
```
Comment (cormacpayne) [inline src/Network/Network.Test/ScenarioTests/ApplicationGatewayTests.ps1]: @EvgenyAgafonchikov just to clarify, is this a scenario that no longer works?

Comment (EvgenyAgafonchikov) [reply src/Network/Network.Test/ScenarioTests/ApplicationGatewayTests.ps1]: @cormacpayne, this scenario should still work.

Previous approach dropped IDs from all the subresources (e.g. FrontendPort) ApplicationGateway was created with, so we had to re-request them to refer again. After updates in this PR IDs are kept, so re-requesting is no more required. This also repros scenario from the related issue #3800

**Pull request merged**